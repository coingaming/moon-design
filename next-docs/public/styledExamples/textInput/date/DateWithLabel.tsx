import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput
      type="date"
      inputSize="medium"
      label="medium"
      placeholder="Placeholder"
    />
    <TextInput
      type="date"
      inputSize="large"
      label="large"
      placeholder="Placeholder"
    />
    <TextInput
      type="date"
      inputSize="xlarge"
      label="xlarge"
      placeholder="Placeholder"
    />
  </div>
);

export default Example;
