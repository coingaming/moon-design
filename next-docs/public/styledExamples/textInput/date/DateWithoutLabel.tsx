import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput type="date" inputSize="medium" placeholder="medium" />
    <TextInput type="date" inputSize="large" placeholder="large" />
    <TextInput type="date" inputSize="xlarge" placeholder="xlarge" />
  </div>
);

export default Example;
