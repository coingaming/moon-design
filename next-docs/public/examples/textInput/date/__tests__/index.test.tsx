/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';

import DateWithLabel from '../DateWithLabel';
import DateWithoutLabel from '../DateWithoutLabel';
import DateWithHint from '../DateWithHint';
import DateDisabled from '../DateDisabled';
import DateError from '../DateError';
import DateBgColor from '../DateBgColor';
import DateReadonly from '../DateReadonly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Date', () => {
  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateWithLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateWithoutLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateWithHint />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateDisabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateError />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateBgColor />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(renderWithTheme(<DateReadonly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
