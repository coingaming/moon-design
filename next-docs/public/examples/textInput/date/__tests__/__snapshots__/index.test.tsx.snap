// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Date renders disabled 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        disabled={true}
        placeholder=" "
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        disabled={true}
        placeholder=" "
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          disabled={true}
          placeholder=" "
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders error 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 kwpJwN"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 dSfsRu"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 gKjnuf"
          placeholder="Placeholder"
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders readOnly 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder=" "
        readOnly={true}
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder=" "
        readOnly={true}
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder=" "
          readOnly={true}
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders with hint 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="Placeholder"
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders with label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="Placeholder"
        type="date"
      />
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="Placeholder"
        type="date"
      />
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="Placeholder"
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders without label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="medium"
        type="date"
      />
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="large"
        type="date"
      />
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 kwsobc"
          placeholder="xlarge"
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date supports background colour 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 cgHTJu"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 hAmOHl"
        placeholder="Placeholder"
        type="date"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4 w-52"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 hYWLpt"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="Placeholder"
          type="date"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;
